packages/core/src/connectors/index.ts

```diff
@@ -99,15 +94,16 @@ export const getConnectorInstanceByType = async <T extends ConnectorInstance>(
 };
 
 export const initConnectors = async () => {
+  const connectors = await findAllConnectors();
+  const existingConnectorIds = new Set(connectors.map((connector) => connector.id));
+
   await Promise.all(
     allIConnectors.map(async ({ metadata: { id } }) => {
-      if (await hasConnector(id)) {
+      if (existingConnectorIds.has(id)) {
```

Comment:
It would be more clear?

```
if (connectors.find(({ id }) => id)) {
```

Though your code has better performance. @logto-io/eng

Reply:
if we know there will be dozens of connectors, it does not seem to be a good idea？

Reply:
For me, is OK to have depth-2 loops, maybe take a discussion?